mono/metadata/threads.c

```diff
@@ -257,8 +257,14 @@ mono_thread_get_tls_key (void)
 gint32
 mono_thread_get_tls_offset (void)
 {
-	int offset;
+	int offset = -1;
+
+#ifdef HOST_WIN32
```

Comment:
Oh, I see why.

We should clean up MONO_THREAD_VAR_OFFSET on windows to follow the same idiom as OSX and have MONO_THREAD_VAR_OFFSET do the trick.